src/arrayMethodJoin.js

```diff
@@ -4,8 +4,26 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let stringFromArray = '';
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === this.length - 1 && typeof this[i] !== 'undefined') {
+        stringFromArray += `${this[i]}`;
+
+        return stringFromArray;
+      } else if (i === this.length - 1) {
+        return stringFromArray;
+      }
```

Comment:
No need to make checks for last element and return. Your `for` loop will end after last element. You have the return in the bottom of your function. Also, you can handle your comma logic once below, no need for special cases

```suggestion
```